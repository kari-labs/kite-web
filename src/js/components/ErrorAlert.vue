<template>
    <v-alert :value="alert" type="error" icon="warning" transition="slide-y-transition">
        {{text}}
    </v-alert>
</template>

<script>
    export default {
        props: [
            'alert',
            'text'
        ]
    }
</script>

<style lang="stylus" scoped>
    .v-alert.error
        margin-top 0px
</style>