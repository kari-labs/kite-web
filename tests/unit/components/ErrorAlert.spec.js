import { mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import ErrorAlert from '@/components/ErrorAlert.vue'

describe('ErrorAlert', () => {
    let wrapper

    beforeEach(() => {
        Vue.use(Vuetify)

        // Fixes Vuetify nag
        document.body.setAttribute('data-app', true)
        wrapper = mount(ErrorAlert, {
            propsData: {
                alert: true,
                text: 'foobar'
            }
        })
    })

    it('should render Vuetify components', () => {
        expect(wrapper.contains('div.v-alert')).toBe(true)
    })

    it('should render alert correctly', () => {
        wrapper.props.text = 'foo bar foobar BarFoo'
        expect(wrapper.element).toMatchSnapshot()
    })

    it('should render correctly when hidden', () => {
        wrapper = mount(ErrorAlert, {
            propsData: {
                alert: false,
                text: 'foo and also bar'
            }
        })

        Vue.nextTick(() => {
            expect(wrapper.element).toMatchSnapshot()
        })
    })
})