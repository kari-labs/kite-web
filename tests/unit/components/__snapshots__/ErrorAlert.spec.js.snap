// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert should render alert correctly 1`] = `
<div
  class="v-alert error"
>
  <i
    aria-hidden="true"
    class="v-icon material-icons theme--light v-alert__icon"
  >
    warning
  </i>
  <div>
    
    foobar

  </div>
</div>
`;

exports[`ErrorAlert should render correctly when hidden 1`] = `
<div
  class="v-alert error"
  style="display: none;"
>
  <i
    aria-hidden="true"
    class="v-icon material-icons theme--light v-alert__icon"
  >
    warning
  </i>
  <div>
    
    foo and also bar

  </div>
</div>
`;
